src/test/kotlin/no/nav/syfo/sykmelding/model/SykmeldingMapperKtTest.kt

```diff
@@ -28,161 +27,257 @@ class SykmeldingMapperKtTest : Spek({
-                fom = LocalDate.of(2020, 3, 10),
-                tom = LocalDate.of(2020, 3, 20)
-            )))
+            val sykmeldingDbModel = getSykmeldingerDBmodel(
+                skjermet = true,
+                perioder = listOf(
+                    getPeriode(
+                        fom = LocalDate.of(2020, 3, 10),
+                        tom = LocalDate.of(2020, 3, 20)
+                    )
+                )
+            )
             val sykmeldingMedUtdypendeOpplysninger = sykmeldingDbModel.copy(
-                sykmeldingsDokument = sykmeldingDbModel.sykmeldingsDokument.copy(utdypendeOpplysninger = objectMapper.readValue(utdypendeopplysningerJson))
+                sykmeldingsDokument = sykmeldingDbModel.sykmeldingsDokument.copy(
+                    utdypendeOpplysninger = objectMapper.readValue(
+                        utdypendeopplysningerJson
+                    )
+                )
             )
 
-            val mappetSykmelding = sykmeldingMedUtdypendeOpplysninger.toSykmeldingDTO(sporsmal = emptyList(), isPasient = false, ikkeTilgangTilDiagnose = false)
+            val mappetSykmelding = sykmeldingMedUtdypendeOpplysninger.toSykmeldingDTO(
+                sporsmal = emptyList(),
+                isPasient = false,
+                ikkeTilgangTilDiagnose = false
+            )
 
-            mappetSykmelding.andreTiltak shouldEqual "Andre tiltak"
-            mappetSykmelding.skjermesForPasient shouldEqual true
-            mappetSykmelding.tiltakNAV shouldEqual "Tiltak NAV"
-            mappetSykmelding.medisinskVurdering shouldEqual MedisinskVurderingDTO(hovedDiagnose = DiagnoseDTO("L87", "ICPC-2", "Bursitt/tendinitt/synovitt IKA"), biDiagnoser = emptyList(), annenFraversArsak = null, svangerskap = false, yrkesskade = false, yrkesskadeDato = null)
-            mappetSykmelding.meldingTilNAV shouldEqual MeldingTilNavDTO(true, "Masse bistand")
-            mappetSykmelding.utdypendeOpplysninger shouldEqual objectMapper.readValue(mappedOpplysningerJson)
+            mappetSykmelding.andreTiltak shouldBeEqualTo "Andre tiltak"
+            mappetSykmelding.skjermesForPasient shouldBeEqualTo true
+            mappetSykmelding.tiltakNAV shouldBeEqualTo "Tiltak NAV"
+            mappetSykmelding.medisinskVurdering shouldBeEqualTo MedisinskVurderingDTO(
+                hovedDiagnose = DiagnoseDTO(
+                    "L87",
+                    "ICPC-2",
+                    "Bursitt/tendinitt/synovitt IKA"
+                ),
+                biDiagnoser = emptyList(),
+                annenFraversArsak = null,
+                svangerskap = false,
+                yrkesskade = false,
+                yrkesskadeDato = null
+            )
+            mappetSykmelding.meldingTilNAV shouldBeEqualTo MeldingTilNavDTO(true, "Masse bistand")
+            objectMapper.writeValueAsString(mappetSykmelding.utdypendeOpplysninger) shouldBeEqualTo mappedOpplysningerJson
```

Comment:
Her har jeg på en måte gjort motsatt av hvordan det var før. Selv med gammel versjon av jackson feilet dette, og jeg skjønner ikke hvordan det kan ha virket før.. Feilmeldingen jeg fikk var at den forventet dette: 
<Pairs:
[(6.2,
{6.2.1=SporsmalSvarDTO(sporsmal=spm, svar=svar, restriksjoner=[SKJERMET_FOR_ARBEIDSGIVER]),
6.2.2=SporsmalSvarDTO(sporsmal=spm, svar=svar, restriksjoner=[SKJERMET_FOR_ARBEIDSGIVER]),
6.2.4=SporsmalSvarDTO(sporsmal=spm, svar=svar, restriksjoner=[SKJERMET_FOR_PASIENT])})]>

men fikk dette: <Pairs:
 [(6.2,
 {6.2.1={sporsmal=spm, svar=svar, restriksjoner=[SKJERMET_FOR_ARBEIDSGIVER]},
 6.2.2={sporsmal=spm, svar=svar, restriksjoner=[SKJERMET_FOR_ARBEIDSGIVER]},
 6.2.4={sporsmal=spm, svar=svar, restriksjoner=[SKJERMET_FOR_PASIENT]}})]>

(linjeskiftene er mine for lesbarhet)

Reply:
Kanskje default toString-metode har endret seg ved oppgraderingen av Java?

Reply:
har ikke peiling